mobile/src/components/RegisterBeneficiaryForm/index.js

```diff
@@ -321,8 +321,13 @@ const BeneficiaryDetails = ({verifyDetails, formData, setValue, errors}) => {
                         hidden={verifyDetails}>
                     <option disabled selected={!formData.district} value>Select District</option>
                     {
-                        districts.map(d => <option selected={d.name === formData.district}
-                                                   value={d.name}>{d.name}</option>)
+                        districts.map(d => { 
+                            if(d.name.indexOf(formData.district) >= 0) {
+                                formData.district = d.name;
```

Comment:
formData should be updated using setFormData

Reply:
done